yt_dlp/extractor/vidflex.py

```diff
@@ -0,0 +1,124 @@
+import base64
+import json
+
+from .common import InfoExtractor
+from ..utils import (
+    mimetype2ext,
+    traverse_obj,
+    url_or_none,
+)
+
+
+class VidflexIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:[^/]+)/[a-z]{2}(?:-[a-z]{2})?/c/[\w-]+\.(?P<id>\d+)'
```

Comment:
We should not match such a generic regex. Since a embed extractor is also not really viable, we need to hardcode the supported domains.
I was thinking something like this (untested)
```
    _DOMAINS = [
        'video.hockeycanada.ca',
        'www.figureitoutbaseball.com',
        'videos.telusworldofscienceedmonton.ca',
        # TODO: Add more domains
    ]
    _BASE_URL_RE = rf'https?://(?:[^.]+\.vidflex\.tv|{"|".join(map(re.escape, _DOMAINS))})'
    _VALID_URL = rf'{_BASE_URL_RE}/[a-z]{{2}}(?:-[a-z]{{2}})?/c/[\w-]+\.(?P<id>\d+)'
```
Feel free to add any more domains which you can find. I'll also see if I can find some more.

Reply:
Missing subdomains? I tried this and it works great:

```suggestion
    _DOMAINS_RE = [
        r'[^.]+\.vidflex\.tv',
        r'(?:www\.)?figureitoutbaseball\.com',
        r'(?:www\.)?tuffhedemantv\.com',
        r'(?:www\.)?albertalacrossetv\.com',
        r'(?:www\.)?silenticetv\.com',
        r'video\.hockeycanada\.ca',
        r'videos\.telusworldofscienceedmonton\.ca',
        # TODO: Add more domains
    ]
    _VALID_URL = rf'^https?://(?:{"|".join(_DOMAINS_RE)})/[a-z]{{2}}(?:-[a-z]{{2}})?/c/[\w-]+\.(?P<id>\d+)'
```

Your opinion?

---

> Feel free to add any more domains which you can find. I'll also see if I can find some more.

I've checked all open issues but didn't find any URL matches the above regex.

Reply:
> I've checked all open issues but didn't find any URL matches the above regex.

I've checked all open issues but didn't find any URL matches the path part of the above regex. How do we easily find a website with the Vidflex platform?

Reply:
The code you posted above also works, though not strictly necessary since the www. subdomains redirect. You can use whichever variant you prefer.

I mostly just searched on Google. Here some of the other URLs I found:
- https://www.ocaalive.com/
- https://watch.wpca.com/ (maybe seems to have no working videos, but only livestreams?)
- https://videos.livingworks.net/
- https://live.ofsaa.on.ca/ (only live as well?)
- https://tv.procoro.ca/
- https://cjfltv.com/ (only with account?)
- https://www.praxisseries.ca/ 
- https://www.pegasussports.tv/
- https://www.acactv.ca
- https://tv.realcastmedia.net/
- https://watch.rekindle.tv/
- https://watch.binghamtonbulldogs.com/ (only live?)
- https://video.huuayaht.org/
- https://video.turningpointensemble.ca/
- https://tv.fringetheatre.ca/
- https://www.watchfuntv.com/
- https://video.haisla.ca/

We don't need to find all of them, maybe just add those for now, and we can add more as people request them/find them.

Reply:
Please check da4267b8e3d50d9a2c457ed13d665a78221ee45d.

Reply:
LGTM. I'd remove the todo comment.

Reply:
> remove the todo comment.

done